custom-schemes.md

```diff
@@ -2,40 +2,54 @@
 
-4. What is the user experience when no wallet app is installed or when the user cancels wallet invocation?
+### What is the user experience when no wallet app is installed or when the user cancels wallet invocation?
 
-   - Browsers generally silently ignore navigation to URL schemes with no handler installed. This likely means a website cannot distinguish between nothing happening while waiting for a wallet to process a request and nothing happening because no wallet app is installed. 
+   - Browsers generally silently ignore navigation to URL schemes with no handler installed. This likely means a website cannot distinguish between nothing happening while waiting for a wallet to process a request and nothing happening because no wallet app is installed.
    - Additionally browsers on Android and iOS often offer the user the choice of whether or not to invoke an application for a custom URL scheme navigation (even if there is only one such app). The user canceling at this stage is also indistinguishable from no valid application being installed.
    - A browser-based API could show the user some meaningful error message when no relevant wallets are installed.
 
-5. What is the user experience on a desktop operating system?
+### What is the user experience on a desktop operating system?
 
    - Most wallet applications exist only as mobile applications, and users are generally reluctant if not unable (due to enterprise security settings, for instance) to install native applications on desktop operating systems. What fallback will flows relying on custom URL schemes provide for users of desktop operating systems?
    - A browser-based API can potentially provide low-friction integration between a desktop browser and a wallet application running on a mobile phone or a website that provides an in-browser wallet experience on the same device.
 
-6. What are the privacy implications of a wallet accepting custom schemes?
+### Friction and app switching
+
+Opening a custom URL scheme will switch apps, which is a high friction user flow. A browser API could present relevant UI (even from third party wallet apps) in a sheet over the webpage, which is a less disruptive flow.
+
+## What are the privacy implications of a wallet accepting custom schemes?
 
    - [Android aims](https://developer.android.com/training/package-visibility) to keep the list of installed applications private from other applications, but this capability is reduced for applications which handle arbitrary intents like custom URL schemes. By registering to support custom URL schemes, a wallet application may effectively be leaking PII (such as the user's likely citizenship status or state of residency) to other applications on the device.
```

Comment:
The leak happens only when the website learns whether the app was opened or not (or am I missing anything?). Therefore, having e.g., a user interaction before the wallet calls back to the website (via a backchannel) would help to reduce the impact of this privacy leak. (It cannot be exploited in large scale any longer.)

Reply:
@danielfett see this thread for background on this statement: https://github.com/WICG/identity-credential/pull/37#discussion_r1375474010

Reply:
Yeah that's not the problem. It's that registering an intent filter on Android makes your Android package name discoverable by other Android apps (silently and at any time) which say they need to query the system for apps satisfying that intent filter.